lib/create-environment-api.ts

```diff
@@ -1181,15 +1181,12 @@ export default function createEnvironmentApi({ http }: { http: AxiosInstance })
         .then((data) => wrapSnapshotCollection<ContentTypeProps>(http, data))
     },
 
-    createTag(id: string, name: string, visibility: TagVisibility = 'private') {
+    createTag(id: string, name: string, visibility?: TagVisibility) {
       const raw = this.toPlainObject() as EnvironmentProps
       const params = { spaceId: raw.sys.space.sys.id, environmentId: raw.sys.id, tagId: id }
 
       return endpoints.tag
-        .createWithId(http, params, {
-          name,
-          sys: { visibility },
-        })
+        .createWithId(http, params, { name }, visibility)
```

Comment:
Would it be better to have the last option as a headers object instead?

Reply:
@burakukula Not sure, it would change the API to (if you want to set the visibility via the `plainClient`):
```javascript
.createWithId(http, params, {name}, { 'X-Contentful-Tag-Visibility': visibility })
```

What do you think?

Reply:
Ah, what I had in mind was: `{ visibility }` and then the method knows which header key to use.

Reply:
In the `createWithId` signature, the last param can be named `headers`.

Reply:
Ah I see, so we would have an object with  `{visibility: 'public' | 'private'}` and would transform the key to `X-Contentful-Tag-Visibility`?

Reply:
@makinwab I think I personally prefer to just have the visibility value there, but not a strong opinion :)

Reply:
My comment is non-blocking, actually. Approved.